### PR TITLE
Support creating Datadog connection from the UI and add documentation for this conn

### PR DESCRIPTION
**Pull request by hussein-awala**
related: #28790 

---
This PR:

- add  Datadog connection to webserver connections list
- updates Datadog connection UI widget to easily provide extra values
- add documentation for Datadog provider connection types.

### CODE REVIEW

File: airflow/providers/datadog/hooks/datadog.py [hunk 1]
```diff
@@ -152,3 +157,27 @@ def post_event(
...
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import StringField
+
+        return {
+            "api_host": StringField(lazy_gettext("Datadog API endpoint"), widget=BS3TextFieldWidget()),
+            "api_key": StringField(lazy_gettext("Datadog API key"), widget=BS3TextFieldWidget()),
+            "app_key": StringField(lazy_gettext("Datadog application key"), widget=BS3TextFieldWidget()),
+            "source_type_name": StringField(
+                lazy_gettext("Datadog source type name"), widget=BS3TextFieldWidget()
```
Comment (josh-fell) [inline airflow/providers/datadog/hooks/datadog.py]:
```suggestion
            "api_host": StringField(lazy_gettext("API endpoint"), widget=BS3TextFieldWidget()),
            "api_key": StringField(lazy_gettext("API key"), widget=BS3TextFieldWidget()),
            "app_key": StringField(lazy_gettext("Application key"), widget=BS3TextFieldWidget()),
            "source_type_name": StringField(
                lazy_gettext("Source type name"), widget=BS3TextFieldWidget()
```
WDYT about removing "Datadog" from the labels? This feels a little redundant given users would have chosen the Datadog conn type and the fields are assumed to be Datadog related.

File: docs/apache-airflow-providers-datadog/connections/datadog.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Hooks and sensors related to Datadog use ``datadog_default`` by default.
+
+Configuring the Connection
+--------------------------
+Host
+    Host name to associate with the sent events.
+
+Extra
+    Specify the extra parameters (as json dictionary) that can be used in Datadog
```
Comment (josh-fell) [inline docs/apache-airflow-providers-datadog/connections/datadog.rst]:
```suggestion
    Specify the extra parameters (as a JSON dictionary) that can be used in Datadog
```
A nit I suppose.

**Review by josh-fell — APPROVED** [review]
LGTM; just small comments.

File: airflow/providers/datadog/hooks/datadog.py (hunk 1 above)
Comment (hussein-awala) [reply airflow/providers/datadog/hooks/datadog.py]: I had the same impression :+1:

**Review by potiuk — APPROVED** [review]

**Pull request merged**